code/modules/mob/living/carbon/life.dm

```diff
@@ -28,7 +28,7 @@
 
 	if(stat == DEAD)
 		stop_sound_channel(CHANNEL_HEARTBEAT)
-		LoadComponent(/datum/component/rot/corpse)
+		SEND_SIGNAL(src, COMSIG_START_MIASMA_GENERATION)
```

Comment:
Signals are supposed to have a single source, what is this doing here?

Reply:
See above

Reply:
This event should happen on death, no? When the `stat` changes to `DEAD`?

Reply:
Is there a signal for that or something? A better way to hook into it then this?

Reply:
`COMSIG_MOB_STATCHANGE`